File: src/0070-ClimbStairs/climbStairs.cpp
#include <math.h>
#include <iostream>

using namespace std;

// fibonacci 
int climbStairs(int n) {
    int p = 0, q = 0, r = 1;
    for (int i = 1; i <= n; i++)
    {
        p = q;
        q = r;
        r = p + q;
    }
    return r;
}

int climbStairs(int n) {
    double sqrt5 = sqrt(5);
    double fibn = pow((1 + sqrt5) / 2, n + 1) - pow((1 - sqrt5) / 2, n + 1);
    return (int)(fibn / sqrt5);
}

int main()
{
    cout << climbStairs(0) << endl;
    cout << climbStairs(10) << endl;
    return 0;
}

File: src/0070-ClimbStairs/ClimbStairs.swift

// fibonacci number sequence
func climbStairs(_ num: Int) -> Int {
    if num <= 1 {
        return 1
    }

    var f1 = 1
    var f2 = 1
    var tmp = f2
    for i in 2...num {
        f2 = (f1 + f2)
        f1 = tmp
        tmp = f2
    }
    return f2

    // return climbStairs(n-1) + climbStairs(n-2)
}

print(climbStairs(20))